### DISCUSSION TITLE
Persistent "Accessed by Other Users" Error with Prisma Migrate on Vercel Postgres

### DISCUSSION THREAD

**Discussion by jocruz** [prisma/prisma#21859] [Q&A] [answered]
### Question

Hello Prisma Community,

I'm experiencing a persistent issue when trying to run migrations using Prisma with a PostgreSQL database hosted on Vercel. Despite successfully terminating other sessions using the PostgreSQL pg_terminate_backend() function, I keep encountering the following error:

```

Error: ERROR: database "prisma_migrate_shadow_db_a016aad6-2481-4191-a738-ffcdd61ce3a1" is being accessed by other users
DETAIL: There are 2 other sessions using the database.
   0: schema_core::state::DevDiagnostic
             at schema-engine\core\src\state.rs:270
```

This error suggests that the shadow database used by Prisma during migration is accessed by other users or processes, which is blocking the migration.

Here are the steps I've already taken:

- Attempted to terminate other sessions to the shadow database with the `pg_terminate_backend()` function. However I am limited since Vercel storage dashboard query and pgadmin will not let me run a terminate session query
- Ensured that `await prisma.$disconnect()` is called in my code after operations to close the Prisma Client connection. I did not have this on my code however I added it afterwards trying to run the .findMany() 
- Attempted to restart the migration process after some delay, seems like no matter how long I wait and the more migrations I attempt it keeps creating shadow databases that I can access but not seem to delete or disconnect from.

Has anyone encountered this issue and found a solution, or can provide insights on how to troubleshoot this further?

attempted to run the commands 
` npx prisma migrate dev --name init
` and 
`  npx prisma migrate dev --name add-total-players-column
`

I am using Vercel Storage postgres
I am using Pg admin 4

I am using next.js and vscode

Any advice or suggestions would be greatly appreciated!

Thank you!

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by janpio** [comment, marked as answer]
Which connection string are you using from Vercel Postgres?
The direct one, or the connection pooled on?
For migrations you should optimally use the direct one. The connection pooler can keep connections to the temporary shadow database open and cause error message like the one you reported.

**Reply by s1xp4c** [reply]
Fixed it for me, migrating initial Prisma scema - Thanx :)

**Reply by GCinellu** [reply]
I had the same issue and was, in fact, connected to the pooler. Thank you so much.

**Reply by erik-stodola** [reply]
I had to end up using `DEV_URL_NON_POOLING` instead of `DEV_PRISMA_URL`

**Reply by RejownAhmed** [reply]
Yes this is the solution in the most of the cases. We have to disable pooling and select the direct connection string.

**Comment by ggaabe** [comment]
getting the same issue now with the recent migration to supavisor. every time I manually terminate a session a new shadow DB ID appears that a user session remains connected to when trying to run the migration

**Comment by samuelpalacioss** [comment]
Now with the supabase update, I'm getting the same issue

**Reply by teslansoft** [reply]
Same for me I'm getting the same error when I try to run my migration using Prisma with supabase

**Reply by ludralph** [reply]
Hi @samuelpalaciosdev and @teslansoft, can you please open a new discussion using the discussion template to describe your issue?

**Reply by GCinellu** [reply]
Try to see if the answer above applies to Supabase too.

https://github.com/prisma/prisma/discussions/21859#discussioncomment-7520072

EDIT: by checking their docs, Supabase does in fact offer multiple options. Make sure to try invoking the migration with the direct connection string as for the Vercel solution.
https://supabase.com/docs/guides/database/connecting-to-postgres
